packages/transport/src/chrome-runtime/background/connection.ts

```diff
@@ -0,0 +1,151 @@
+import { UniversalServerRequest } from '@connectrpc/connect/protocol';
+import {
+  ChannelConfig,
+  ChannelConfigString,
+  ChannelLabel,
+  parseConnectionName,
+} from '../../channel/names';
+import { ChromeRuntimeConnectionListener, ChromeRuntimePortListener } from '../types';
+import { JsonValue } from '@bufbuild/protobuf';
+import { TransportMessage } from '../../channel/control';
+
+interface OriginRegistry {
+  known(origin: string): Promise<boolean>;
+  //services(origin: string): Promise<Record<string, false | ChromeRuntimeAdapterOptions>>;
+  services(origin: string): Promise<Record<string, (x: JsonValue) => Promise<JsonValue>>>;
```

Comment:
this would be an authentication/permission gateway, but it was convenient to adapt here into the existing router.